x-pack/plugins/security_solution/server/endpoint/routes/metadata/handlers.ts

```diff
@@ -277,6 +309,10 @@ export async function enrichHostMetadata(
       log.warn(`Missing elastic agent id, using host id instead ${elasticAgentId}`);
     }
 
+    if (metadataRequestContext.requestHandlerContext == null) {
```

Comment:
rollback

Reply:
I change my mind, in order to be compatible with the case when metadataRequestContext.requestHandlerContext may not exist, I'd like to keep the logic here and check one of them (metadataRequestContext.esClient and metadataRequestContext.requestHandlerContext?.core.elasticsearch.client) should exist.